src/loader/ident_env.cppo.ml

```diff
@@ -659,22 +655,17 @@ let find_extension_identifier env id =
 let find_value_identifier env id =
   Ident.find_same id env.values
 
+(** Lookup a type in the environment. If it isn't found, it's assumed to be a
+    core type. *)
```

Comment:
That means we are confident there is no error in our building of the environment.
Would it make sense to include a simple check that `Ident.name id` is part of a list of predefined types? Kind of similar to what was before, but much more lightweight and easily removable.

Reply:
This patch is intended to be used with patched compilers and not in prevision of new core types. Not having a check is intended.

To more easily debug wrongly built environments, we'd just need to differentiate core types from types that didn't resolve in the generated HTML. We could wrap core types in a `span` with a class (they don't need to be styled differently for this purpose but could be) but I think we can already debug this case without this.

Reply:
I agree that the check is somehow redundant with OCaml having successfully compiled the file, and that it will be annoying for patched compilers.

Reply:
Thanks for reviewing :) Now that we agree, I'll merge!